testresultparser/parser_test.go

```diff
@@ -0,0 +1,154 @@
+package testresultparser_test
+
+import (
+	"path"
+	"path/filepath"
+	"runtime"
+
+	. "github.com/almighty/almighty-test-runner/testresultparser"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+var _ = Describe("Parser", func() {
+
+	Describe("Parsing XML file", func() {
+		Context("With Maven-surefire plugin and Junit", func() {
+			It("should return output struct", func() {
+				// given
+				f := []TestFailure{{TestCase: "testApp", Type: "junit.framework.AssertionFailedError", Message: "junit.framework.AssertionFailedError\n\tat junit.framework.Assert.fail(Assert.java:47)\n\tat junit.framework.Assert.assertTrue(Assert.java:20)\n\tat junit.framework.Assert.assertTrue(Assert.java:27)\n\tat org.hemani.javabrains.AppTest.testApp(AppTest.java:36)\n"}}
+				expectedResult := &TestResult{TestSuite: "org.hemani.javabrains.AppTest", Tests: 1, Failures: 1, Errors: 0, Skipped: 0, Time: "0.017", Failure: f}
+
+				// when
+				filepath := getFilepath("junit_failure_test.xml")
```

Comment:
I would consider it as 'initial context' or 'test fixture' and hence move it to `given` block.